packages/substyle/src/useStyles.js

```diff
@@ -25,110 +26,28 @@ type InternalOverrideT = {|
 
 type Args = [StyleT, OverrideT] | [StyleT, OverrideT, ModifiersT]
 
-const useStyles = (...args: Args) => {
-  const [defaultStyle, { style, className, classNames }, modifiers] = parseArgs(
-    ...args
-  )
-
+const useStyles = (
+  defaultStyle?: StyleT,
+  { style, className, classNames }: OverrideT,
+  modifiers?: ModifiersT
+) => {
   const propsDecorator = useContext(PropsDecoratorContext)
 
   const substyle = useMemo(
     () => createSubstyle({ style, className, classNames }, propsDecorator),
     [style, className, classNames, propsDecorator]
   )
 
-  return useMemo(() => substyle(modifiers, defaultStyle), [
+  // Modifiers are recalculated on every render. To leverage memoization we need to map the
+  // selected modifiers (array of strings) to something that keeps its identity if the set of
+  // modifiers does not change.
+  // We join them to a string, assuming users won't change the order of keys in the modifiers object.
+  const selectedModifiers = coerceSelection(modifiers)
+  return useMemo(() => substyle(selectedModifiers, defaultStyle), [
     substyle,
-    modifiers,
+    selectedModifiers.join(','),
     defaultStyle,
   ])
 }
```

Comment:
Do we need this? Because the `substyle` method itself does memoization. Isn't there even a test that should fail if this wouldn't work?

Reply:
That is a very good point. It looks like we don't need it.